### ISSUE TITLE
Update Dual List Selector to work in screen reader

### ISSUE THREAD

**Issue by jessiehuff** [patternfly/patternfly-react#5149]
**Describe the issue. What is the expected and unexpected behavior?**
Currently the dual list selector doesn't work in VO or JAWS. 
My guess is the `<li><button>` structure is causing issues:

```<div role="listbox">
  <ul>
    <li role="option">
      <button>
```
For focus management to work, the element with `role="option"` should be getting focus, but it looks like the button is what gets focus. Also, the element with `role="listbox"` should have an `aria-describedby` that points to the text stating how many items are selected (i think, assuming this isn’t announced automatically with appropriate aria attributes).
For managing focus, I wonder if it’s a cleaner experience to use `aria-activedescendent`, and keep focus always on the `role="listbox"`.

I think part of the issue is the core example uses buttons in the list box. The WAI-ARIA example has just a `<ul><li>` DOM structure. I haven’t looked into why all of these `<span>`s are nested inside the button, if they’re needed for styling purposes, but I think we should try to match what the WAI-ARIA example has. So this would mean:
moving attributes from the parent div to the child ul, and make sure any other aria attributes needed for the ul are included
changing the button to a span, and moving it’s attributes to the parent li, and adding any other aria attributes needed for the li

However, all of these theories should be tested. We should adapt the aria attributes and the structure and see if we can get this to function in a screen reader.

**Please provide the steps to reproduce. Feel free to link CodeSandbox or another tool.**

Open the dual list selector in VO and attempt to use the dual list selector.

**Is this a bug or enhancement? If this issue is a bug, is this issue blocking you or is there a work-around?**
Bug

**Comment by tlabaj** [comment]
cc @mcoker

**Comment by nicolethoen** [comment]
The tree view implementation of the dual list selector does not use a button in the `<li>` at all. At the very least consistency should be considered. That will probably help with accessibility and consistent interaction.
